go/tasks/plugins/array/k8s/transformer.go

```diff
@@ -94,6 +94,40 @@ func buildPodMapTask(task *idlCore.TaskTemplate, metadata core.TaskExecutionMeta
 	return pod, nil
 }
 
+// Here we customize the k8sPod primary container by templatizing args.
+// The call to ToK8sPodSpec for the task container target
+// case already handles this but we must explicitly do so for K8sPod task targets.
+func modifyMapPodTaskPrimaryContainer(ctx context.Context, tCtx core.TaskExecutionContext, arrTCtx *arrayTaskContext, container *v1.Container) error {
+	var err error
+	container.Args, err = template.Render(ctx, container.Args,
+		template.Parameters{
+			TaskExecMetadata: tCtx.TaskExecutionMetadata(),
+			Inputs:           arrTCtx.arrayInputReader,
+			OutputPath:       tCtx.OutputWriter(),
+			Task:             tCtx.TaskReader(),
+		})
+	if err != nil {
+		return err
+	}
+	container.Command, err = template.Render(ctx, container.Command,
+		template.Parameters{
+			TaskExecMetadata: tCtx.TaskExecutionMetadata(),
+			Inputs:           arrTCtx.arrayInputReader,
+			OutputPath:       tCtx.OutputWriter(),
+			Task:             tCtx.TaskReader(),
+		})
+	if err != nil {
+		return err
+	}
+	resources := flytek8s.ApplyResourceOverrides(ctx, container.Resources)
```

Comment:
Do we apply that only on the primary container? or should it apply to all containers? I don't know what we do in regular pod tasks so the same question applies....

Reply:
good q: we only do it for primary containers in pod tasks now. not sure if it should be extended to other container definitions or if a hands-off model is okay